server/test_common/commonServerSteps.ts

```diff
@@ -83,4 +119,83 @@ When(
   })
 );
 
+When(
+  /^I make a WebSocket connection request to '(.+)'$/,
+  stepWithWorld(async (world, endpoint) => {
+    const { server } = world;
+
+    // Set up the websocket
+    await server.get(`/websockettest${endpoint}`).then(
+      (res) => {
+        expect(res.status).toBe(101);
+      },
+      (err) => {
+        throw err;
+      }
+    );
+  })
+);
+
+Then(
+  /^the WebSocket has received (\d+) messages$/,
+  stepWithWorld(async (world, messageCount) => {
+    const { websocket } = world;
+    if (websocket.onmessage != null) {
+      expect((<jest.Mock>websocket.onmessage).mock.calls.length).toBe(
+        parseInt(messageCount as string)
+      );
+    } else {
+      fail('Expected websocket.onmessage to be mocked');
+    }
+  })
+);
+
+And(
+  'I close the WebSocket',
+  stepWithWorld(async (world) => {
+    const { websocket } = world;
+    websocket.close(0, 'Test close');
+  })
+);
+
+And(
+  'the WebSocket is closed',
+  stepWithWorld(async (world) => {
+    const { websocket } = world;
+    if (websocket.onclose != null) {
+      expect((<jest.Mock>websocket.onclose).mock.calls.length).toBe(1);
```

Comment:
`toHaveBeenCalled` and related - https://jestjs.io/docs/en/expect#tohavebeencalled - is a neater API than `mock.calls.x`

Reply:
Forgot all about the `toHaveBeenCalled*` API.  :) Updated